lxd/instance/instance_utils.go

```diff
@@ -140,6 +140,33 @@ func ValidConfig(sysOS *sys.OS, config map[string]string, profile bool, expanded
 			return fmt.Errorf("Image keys can only be set on instances")
 		}
 
+		if !profile {
+			for specificInstType, specificKeys := range specificInstTypeConfigKeys {
```

Comment:
@my5t3ry if I'm reading this right, for non-profile validation, this won't allow an instance to use the `instancetype.Any` profile keys (which should be allowed for any type of instance).